scripts/devenv-builder/configure-vm.sh

```diff
@@ -216,7 +222,11 @@ else
 fi
 
 if ${BUILD_AND_RUN}; then
-    "${DNF_RETRY}" "localinstall" "$(ls -1 ~/microshift/_output/rpmbuild/RPMS/*/*.rpm)"
+    rpms="$(ls -1 ~/microshift/_output/rpmbuild/RPMS/*/*.rpm)"
+    if ! ${OPTIONAL_RPMS}; then
+        rpms="$(echo "${rpms}" | grep -E 'microshift-(4|greenboot|selinux|networking|release-info)')"
+    fi
+    "${DNF_RETRY}" "localinstall" "${rpms}"
```

Comment:
Could we just install the "microshift" RPM and let its dependencies bring in the others? Or do we have to specify them all on the command line because we're installing from local files?

Reply:
It's the latter - if we'd do `dnf install ./some/path/microshift-4.16.rpm` then it wouldn't know about other RPMs unless we'd put them into a repository

Reply:
Makes sense. It would be cool if we could derive this info from the RPM or its spec file instead of hard-coding it. For example, if the RPM is built already, this command gives you a list of the dependencies:

```
$ rpm -q _output/rpmbuild/RPMS/aarch64/microshift-4.16.0_0.nightly_arm64_2024_01_15_061927_20240124205332_e27825c8b-1.el9.aarch64.rpm --requires
```

and that can be filtered to the packages we build

```
$ rpm -q _output/rpmbuild/RPMS/aarch64/microshift-4.16.0_0.nightly_arm64_2024_01_15_061927_20240124205332_e27825c8b-1.el9.aarch64.rpm --requires | grep '^microshift' | cut -f1 -d=
microshift-greenboot
microshift-networking
microshift-selinux
```

Reply:
I think the best way may be to let `dnf` calculate the dependencies itself. 

```
createrepo _output/rpmbuild
sudo dnf install -y _output/rpmbuild/*/*/microshift-4.*.rpm --repofrompath=microshift-local,./_output/rpmbuild/ 
```